mqtt/mqtt3/tests/common/mod.rs

```diff
@@ -100,14 +101,15 @@ impl mqtt3::IoSource for IoSource {
             // If the connection broke while there were still steps remaining in the TestConnection, then the dropped sender will cause the test
             // to receive a futures_channel::oneshot::Canceled error, so the test will panic before this deadline elapses anyway.
             Box::pin(async {
-                let () = tokio::time::delay_for(std::time::Duration::from_secs(5)).await;
+                tokio::time::delay_for(std::time::Duration::from_secs(5)).await;
                 unreachable!();
             })
         }
     }
 }
 
 /// A single connection between a client and a server
+#[allow(clippy::type_complexity)]
```

Comment:
As discussed, you should be able to move this to the specific field that it complained about instead of applying it to the whole struct.